### PR TITLE
fix(typeahead): Clicking an item does not raise the typeaheadOnSelectevent

### PR DESCRIPTION
**Pull request by koodikindral**
Fixes: #363

### CODE REVIEW

File: components/typeahead/typeahead-container.component.ts [hunk 1]
```diff
@@ -10,7 +10,8 @@ const TEMPLATE:any = {
   [Ng2BootstrapTheme.BS4]: `
   <div class="dropdown-menu"
       [ngStyle]="{top: top, left: left, display: display}"
-      style="display: block">
+      style="display: block"
+      (mouseleave)="hideMenu()">
```
Comment (valorkin) [inline components/typeahead/typeahead-container.component.ts:14]: @koodikindral what if it is selected by pressing tab?

**Comment by slintes** [comment]
Please also see my PR https://github.com/valor-software/ng2-bootstrap/pull/395. I keep the blur handling, and fixed the mouseclick handling by listening to the mousedown event instead click event. This this will also hide the popup for keyboard-only users (which will not work with this PR afaik).

**Comment by Toktik** [comment]
#395 fixes the problem

**Comment by fkowal** [comment]
Fixed using a different approach https://github.com/valor-software/ng2-bootstrap/pull/398. Which one makes more sense ?

**Comment by slintes** [comment]
I like your approach even more than mine :)

**Comment by valorkin** [comment]
Thanks for PR, closed in favor of #398

**Pull request closed without merging**